From 8bb0ca0ade05dcb840d3235402240067e5beb9e4 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 15 Sep 2020 16:18:28 +0700
Subject: [PATCH 195/443] arm64: dts: renesas: Add HiHope RZ/G2M[N] Rev.3.0/4.0
 board with mipi-2.1

This commit adds support for mipi-2.1 for HiHope RZ/G2M[N] Rev.3.0/4.0
board:
- HiHope RZ/G2M Rev.3.0/4.0 board is similar with RZ/G2M Rev.2.0.
- HiHope RZ/G2N Rev.3.0/4.0 board is similar with RZ/G2N Rev.2.0 but
it can support CSI40 IF. This CSI is linked with OV5645 camera sensor
which is controlled by I2C channel 3.

However, disable CSI40 by default to ensure working of VIN devices
for HiHope RZ/G2N Rev.3.0/4.0 board.
Reason:
- VIN/CSI2 can work normally when connect 2 cameras in 2 CSI2-IF
(CSI20, CSI40) at the same time.
- VIN/CSI2 can not work when disconnect 1 camera in CSI20 or CSI40.

Therefore, please delete the commenting out at status property if want
to use 2 cameras at the same time.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/Makefile               |   6 +-
 .../renesas/r8a774a1-hihope-rzg2m-ex-mipi-2.1.dts  |  77 +++++++++++++
 .../renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts  | 120 +++++++++++++++++++++
 3 files changed, 201 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-ex-mipi-2.1.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index 9294436..4aa0eba 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -5,14 +5,16 @@ dtb-$(CONFIG_ARCH_R8A774A1) += r8a774a1-hihope-rzg2m-rev2.dtb \
 			       r8a774a1-hihope-rzg2m-rev2-ex-mipi-2.1.dtb
 dtb-$(CONFIG_ARCH_R8A774A1) += r8a774a1-hihope-rzg2m.dtb \
 			       r8a774a1-hihope-rzg2m-ex.dtb \
-			       r8a774a1-hihope-rzg2m-ex-idk-1110wr.dtb
+			       r8a774a1-hihope-rzg2m-ex-idk-1110wr.dtb \
+			       r8a774a1-hihope-rzg2m-ex-mipi-2.1.dtb
 dtb-$(CONFIG_ARCH_R8A774B1) += r8a774b1-hihope-rzg2n-rev2.dtb \
 			       r8a774b1-hihope-rzg2n-rev2-ex.dtb \
 			       r8a774b1-hihope-rzg2n-rev2-ex-idk-1110wr.dtb \
 			       r8a774b1-hihope-rzg2n-rev2-ex-mipi-2.1.dtb
 dtb-$(CONFIG_ARCH_R8A774B1) += r8a774b1-hihope-rzg2n.dtb \
 			       r8a774b1-hihope-rzg2n-ex.dtb \
-			       r8a774b1-hihope-rzg2n-ex-idk-1110wr.dtb
+			       r8a774b1-hihope-rzg2n-ex-idk-1110wr.dtb \
+			       r8a774b1-hihope-rzg2n-ex-mipi-2.1.dtb
 dtb-$(CONFIG_ARCH_R8A774C0) += r8a774c0-cat874.dtb r8a774c0-ek874.dtb \
 			       r8a774c0-ek874-idk-2121wr.dtb \
 			       r8a774c0-ek874-mipi-2.1.dtb
diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-ex-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-ex-mipi-2.1.dts
new file mode 100644
index 0000000..dd29188
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-ex-mipi-2.1.dts
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the HopeRun HiHope RZ/G2M
+ * connected with aistarvision-mipi-v2-adapter board
+ *
+ * Copyright (C) 2020 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a774a1-hihope-rzg2m-ex.dts"
+#define MIPI_PARENT_I2C_CSI20 i2c2
+#include "aistarvision-mipi-adapter-2.1.dtsi"
+
+/ {
+	model = "HopeRun HiHope RZ/G2M with sub board with aistarvision-mipi-v2-adapter board";
+	compatible = "hoperun,hihope-rzg2-ex", "hoperun,hihope-rzg2m",
+		     "renesas,r8a774a1";
+};
+
+&vin0 {
+	status = "okay";
+};
+
+&vin1 {
+	status = "okay";
+};
+
+&vin2 {
+	status = "okay";
+};
+
+&vin3 {
+	status = "okay";
+};
+
+&vin4 {
+	status = "okay";
+};
+
+&vin5 {
+	status = "okay";
+};
+
+&vin6 {
+	status = "okay";
+};
+
+&vin7 {
+	status = "okay";
+};
+
+&csi20 {
+	status = "okay";
+
+	ports {
+		port {
+			csi20_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi20_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi20 {
+	enable-gpios = <&gpio6 4 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio6 8 GPIO_ACTIVE_LOW>;
+
+	port {
+		ov5645_csi20_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi20_in>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts
new file mode 100644
index 0000000..bc8295b
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the HopeRun HiHope RZ/G2N
+ * connected with aistarvision-mipi-v2-adapter board
+ *
+ * Copyright (C) 2020 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a774b1-hihope-rzg2n-ex.dts"
+#define MIPI_PARENT_I2C_CSI20 i2c2
+#define MIPI_PARENT_I2C_CSI40 i2c3
+#include "aistarvision-mipi-adapter-2.1.dtsi"
+
+/ {
+	model = "HopeRun HiHope RZ/G2N with sub board with aistarvision-mipi-v2-adapter board";
+	compatible = "hoperun,hihope-rzg2-ex", "hoperun,hihope-rzg2n",
+		     "renesas,r8a774b1";
+};
+
+&vin0 {
+	status = "okay";
+};
+
+&vin1 {
+	status = "okay";
+};
+
+&vin2 {
+	status = "okay";
+};
+
+&vin3 {
+	status = "okay";
+};
+
+&vin4 {
+	status = "okay";
+};
+
+&vin5 {
+	status = "okay";
+};
+
+&vin6 {
+	status = "okay";
+};
+
+&vin7 {
+	status = "okay";
+};
+
+&csi20 {
+	status = "okay";
+
+	ports {
+		port {
+			csi20_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi20_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi20 {
+	enable-gpios = <&gpio6 4 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio6 8 GPIO_ACTIVE_LOW>;
+
+	port {
+		ov5645_csi20_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi20_in>;
+		};
+	};
+};
+
+&pfc {
+	i2c3_pins: i2c3 {
+		groups = "i2c3";
+		function = "i2c3";
+	};
+};
+
+&i2c3 {
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&csi40 {
+	/* Remove below commenting out if want to use 2 cameras */
+	/* status = "okay"; */
+
+	ports {
+		port {
+			csi40_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi40_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi40 {
+	enable-gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+
+	port {
+		ov5645_csi40_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi40_in>;
+		};
+	};
+};
-- 
2.7.4

